merlin/models/tf/blocks/optimizer.py

```diff
@@ -289,18 +316,26 @@ def weights(self) -> List[tf.Variable]:
         """Returns the optimizer's variables."""
         weights = []
         for optimizer_blocks in self.optimizers_and_blocks:
-            weights += optimizer_blocks.optimizer.weights
+            optimizer = optimizer_blocks.optimizer
+            if hasattr(optimizer, "weights"):  # Tensorflow < 2.11
+                weights += optimizer_blocks.optimizer.weights
+            elif hasattr(optimizer, "variables") and callable(
+                optimizer.variables
+            ):  # Tensorflow >= 2.11
+                weights += optimizer_blocks.optimizer.variables()
```

Comment:
if `variables()` was present in earlier versions of tensorflow (2.8, 2.9, 2.10) and equivalent to `.weights` ? then we might be able to simplify to be only this line?